apps/spruce/src/pages/waterfall/WaterfallGrid.tsx

```diff
@@ -21,9 +23,26 @@ type WaterfallGridProps = {
   projectIdentifier: string;
 };
 
+const tracer = trace.getTracer(TRACER_NAME);
+
 export const WaterfallGrid: React.FC<WaterfallGridProps> = ({
   projectIdentifier,
 }) => {
+  let resolveRender: (v?: any) => void;
+  const renderPromise = new Promise((resolve) => {
+    resolveRender = resolve;
+  });
+
+  useEffect(() => {
+    resolveRender();
+  }, []);
+
+  tracer.startActiveSpan("render waterfall", async (span) => {
+    renderPromise.then(() => {
+      span.end();
+    });
+  });
```

Comment:
i wonder if it might make sense to extract this logic into a hook like `useWaterfallTrace` (or something like that) so that the logic is isolated elsewhere!